core/templates/pages/classroom-page/classroom-page.component.ts

```diff
@@ -122,10 +122,10 @@ export class ClassroomPageComponent implements OnDestroy {
         }
       });
     }, (err) => {
-      // User provided classroom doesnot exist. Redirect to default classroom.
-      this.windowRef.nativeWindow.history.pushState(
-        null, 'classroom', AppConstants.DEFAULT_CLASSROOM_URL_FRAGMENT);
-      this.ngOnInit();
+      // Note to developers:
```

Comment:
Why are we removing this redirection?

Reply:
Currently, 'math' is a default classroom but after this PR, we will be requiring math config in the backend datastore before accessing the math page. Now it is no longer a default one. So for example, if the user provides 'classroom_url_fragment' for 'physics' and if it's config does not exist then they will get a 404 error.

Reply:
Note: I just realised Sean is away. This comment is not a blocker for the PR. It can be addressed later on if required.

Reply:
Yes, thank you

Reply:
Ok to 404 if invalid classroom. That should be the intended behaviour going forward. Thanks for checking!